File: gymgenie-mobile/composeApp/src/androidMain/kotlin/com/asc/gymgenie/feature/food_picker/FoodCategoryImage.kt
package com.asc.gymgenie.feature.food_picker

import com.asc.gymgenie.R
import com.asc.gymgenie.nutrition.FoodCategory

internal fun FoodCategory.categoryDrawable(): Int = when (this) {
    FoodCategory.MEAT -> R.drawable.ic_food_meat
    FoodCategory.FISH -> R.drawable.ic_food_fish
    FoodCategory.DAIRY -> R.drawable.ic_food_milk
    FoodCategory.EGGS -> R.drawable.ic_food_egg
    FoodCategory.GRAINS -> R.drawable.ic_food_cereals
    FoodCategory.LEGUMES -> R.drawable.ic_food_legumes
    FoodCategory.VEGETABLES -> R.drawable.ic_food_vegetables
    FoodCategory.FRUITS -> R.drawable.ic_food_fruits
    FoodCategory.NUTS_SEEDS -> R.drawable.ic_food_nuts
    FoodCategory.OILS -> R.drawable.ic_food_oil
    FoodCategory.OTHER -> R.drawable.ic_food_other
}

internal fun deriveFoodDrawable(dishName: String): Int {
    val lower = dishName.lowercase()
    return when {
        lower.containsAny("курица", "куриц", "грудка", "мяс", "говядин", "свинин", "стейк", "индейк") -> R.drawable.ic_food_meat
        lower.containsAny("рыб", "лосось", "тунец", "форель", "сёмга", "семга") -> R.drawable.ic_food_fish
        lower.containsAny("яйц", "омлет") -> R.drawable.ic_food_egg
        lower.containsAny("молок", "творог", "йогурт", "кефир", "сыр", "сметан") -> R.drawable.ic_food_milk
        lower.containsAny("каш", "овсян", "рис", "гречк", "хлеб", "тост", "паст", "макарон", "спагетт") -> R.drawable.ic_food_cereals
        lower.containsAny("салат", "овощ", "капуст", "помидор", "огурц", "морков", "свекл", "брокколи") -> R.drawable.ic_food_vegetables
        lower.containsAny("фрукт", "яблок", "банан", "апельсин", "ягод") -> R.drawable.ic_food_fruits
        lower.containsAny("орех", "миндал", "фундук", "кешью", "семечк") -> R.drawable.ic_food_nuts
        lower.containsAny("масл", "оливк") -> R.drawable.ic_food_oil
        lower.containsAny("фасол", "горох", "чечевиц", "нут", "бобов") -> R.drawable.ic_food_legumes
        else -> R.drawable.ic_food_other
    }
}

private fun String.containsAny(vararg needles: String): Boolean = needles.any { this.contains(it) }

File: gymgenie-mobile/iosApp/iosApp/MealPlan/MealPlanDetail/FoodCategoryImage.swift
import Foundation
import Shared

func foodCategoryImageName(_ category: FoodCategory) -> String {
    if category == FoodCategory.meat { return "ic_food_meat" }
    if category == FoodCategory.fish { return "ic_food_fish" }
    if category == FoodCategory.dairy { return "ic_food_milk" }
    if category == FoodCategory.eggs { return "ic_food_egg" }
    if category == FoodCategory.grains { return "ic_food_cereals" }
    if category == FoodCategory.legumes { return "ic_food_legumes" }
    if category == FoodCategory.vegetables { return "ic_food_vegetables" }
    if category == FoodCategory.fruits { return "ic_food_fruits" }
    if category == FoodCategory.nutsSeeds { return "ic_food_nuts" }
    if category == FoodCategory.oils { return "ic_food_oil" }
    return "ic_food_other"
}

func deriveFoodImageName(_ dishName: String) -> String {
    let lower = dishName.lowercased()
    if containsAny(lower, "курица", "куриц", "грудка", "мяс", "говядин", "свинин", "стейк", "индейк") { return "ic_food_meat" }
    if containsAny(lower, "рыб", "лосось", "тунец", "форель", "сёмга", "семга") { return "ic_food_fish" }
    if containsAny(lower, "яйц", "омлет") { return "ic_food_egg" }
    if containsAny(lower, "молок", "творог", "йогурт", "кефир", "сыр", "сметан") { return "ic_food_milk" }
    if containsAny(lower, "каш", "овсян", "рис", "гречк", "хлеб", "тост", "паст", "макарон", "спагетт") { return "ic_food_cereals" }
    if containsAny(lower, "салат", "овощ", "капуст", "помидор", "огурц", "морков", "свекл", "брокколи") { return "ic_food_vegetables" }
    if containsAny(lower, "фрукт", "яблок", "банан", "апельсин", "ягод") { return "ic_food_fruits" }
    if containsAny(lower, "орех", "миндал", "фундук", "кешью", "семечк") { return "ic_food_nuts" }
    if containsAny(lower, "масл", "оливк") { return "ic_food_oil" }
    if containsAny(lower, "фасол", "горох", "чечевиц", "нут", "бобов") { return "ic_food_legumes" }
    return "ic_food_other"
}

func foodCategoryFromWireValue(_ value: String?) -> FoodCategory? {
    guard let value = value else { return nil }
    switch value {
    case "MEAT": return FoodCategory.meat
    case "FISH": return FoodCategory.fish
    case "DAIRY": return FoodCategory.dairy
    case "EGGS": return FoodCategory.eggs
    case "GRAINS": return FoodCategory.grains
    case "LEGUMES": return FoodCategory.legumes
    case "VEGETABLES": return FoodCategory.vegetables
    case "FRUITS": return FoodCategory.fruits
    case "NUTS_SEEDS": return FoodCategory.nutsSeeds
    case "OILS": return FoodCategory.oils
    case "OTHER": return FoodCategory.other
    default: return nil
    }
}

private func containsAny(_ haystack: String, _ needles: String...) -> Bool {
    needles.contains { haystack.contains($0) }
}
